extension/content.js

```diff
@@ -102,21 +102,24 @@ function infinitelyMore() {
 function addBlameParentLinks() {
 	$('.blame-sha:not(.js-blame-parent)').each((index, commitLink) => {
 		const $commitLink = $(commitLink);
+
 		const $blameMetaContainer = $commitLink.nextAll('.blame-commit-meta');
 		if ($blameMetaContainer.find('.js-blame-parent').length > 0) {
 			return;
 		}
 
+		const blameTargets = traverseBlameDOM($commitLink);
 		const $blameParentLink = $commitLink.clone();
 		const commitSha = /\w{40}$/.exec(commitLink.href)[0];
+		const href = location.pathname.replace(
+			/(\/blame\/)[^\/]+/,
+			`$1${commitSha}`
+		);
 
 		$blameParentLink
 			.text('Blame ^')
 			.addClass('js-blame-parent')
-			.prop('href', location.pathname.replace(
-				/(\/blame\/)[^\/]+/,
-				`$1${commitSha}${encodeURI('^')}`
-			));
+			.prop('href', `${href}#${blameTargets}`);
```

Comment:
Can't we move the whole functionality for blame parent links to an external file?